app/graphql/mutations/update_squeak.rb

```diff
@@ -3,22 +3,33 @@ class UpdateSqueak < Mutations::BaseMutation
     argument :id, ID, required: true
     argument :report, Boolean, required: false
     argument :nut, Boolean, required: false
+    argument :approved, Boolean, required: false
 
     field :squeak, Types::SqueakType, null: false
 
-    def resolve(id: , report: false, nut: false)
+    def resolve(id: , report: false, nut: false, approved: nil)
       squeak = Squeak.find(id)
       if squeak
-        if report == true
-          squeak.reports += 1
-        elsif nut == true
-          squeak.nuts += 1
-        end
-        squeak.save
+        check_arguments(squeak, report, nut, approved)
       end
       { squeak: squeak }
       rescue ActiveRecord::RecordNotFound => e
         GraphQL::ExecutionError.new(e)
     end
+
+    def check_arguments(squeak, report, nut, approved)
```

Comment:
**suggestion:** Look into one of the many ruby validation libraries out there, or maybe consider using a PORO with `include ActiveModel::Model` to mix in validation logic.